src/test.tsx

```diff
@@ -0,0 +1,216 @@
+import { useState, useEffect } from 'react'
+
+interface User {
+  id: number
+  name: string
+  status: any
+}
+
+const users: User[] = [
+  { id: 1, name: 'João', status: 'active' },
+  { id: 2, name: 'Maria', status: 'inactive' },
+  { id: 3, name: 'Pedro', status: 'active' },
+]
+
+function App() {
+  const [selectedUserId, setSelectedUserId] = useState<number | null>(null)
+  const [count, setCount] = useState(0)
+  const [isVisible, setIsVisible] = useState(true)
+  const [totalActiveUsers, setTotalActiveUsers] = useState(0)
+  
+  useEffect(() => {
+    const handleKeyDown = (e: KeyboardEvent) => {
+      if (e.key === 'Escape') {
+        setSelectedUserId(null)
+      }
+    }
+    
+    window.addEventListener('keydown', handleKeyDown)
+  }, [])
```

Comment:
Todo effeito que tem um evento, listener ou qual função que dependa de um desligamento, ele precisa ser desligado na desmontagem do componente:

artigo explicando sobre: https://refine.dev/blog/useeffect-cleanup/#fetch-requests

```suggestion
  useEffect(() => {
    const handleKeyDown = (e: KeyboardEvent) => {
      if (e.key === 'Escape') {
        setSelectedUserId(null)
      }
    }
    
    window.addEventListener('keydown', handleKeyDown)
    
    return () => {
      //  colocar o remove listener aqui
    }
  }, [])
```